### PR TITLE
Adding FAQs

### LINKED ISSUE #294: Inference needs more documentation
[The inference module's documentation](https://github.com/IllinoisCogComp/illinois-cogcomp-nlp/tree/master/inference) is sparse. We prolly need more details here. 
And more details on [the cutting plane solver](https://github.com/IllinoisCogComp/illinois-cogcomp-nlp/blob/master/inference/src/main/java/edu/illinois/cs/cogcomp/infer/ilp/JLISCuttingPlaneILPSolverGurobi.java) and [beam search](https://github.com/IllinoisCogComp/illinois-cogcomp-nlp/blob/master/inference/src/main/java/edu/illinois/cs/cogcomp/infer/ilp/BeamSearch.java), and possibly usage examples (+ tests).

### PR DESCRIPTION
**Pull request by danyaljj**
Changes 
 - Adds FAQs: Closes #294 
 - Tabularizes/Table-s the list of modules: 

screen shot 2016-11-29 at 12 15 36 am

### CODE REVIEW

File: faq.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Frequently Asked Questions 
+
+ - **Something is broken/Something doesn't work/I get werid results/..** Use Github's issue tracker and open a ticket. We will try to address it as soon as possible.
```
Comment (mssammon) [inline faq.md]: typo -- "werid" -> "wierd"

File: faq.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Frequently Asked Questions 
+
+ - **Something is broken/Something doesn't work/I get werid results/..** Use Github's issue tracker and open a ticket. We will try to address it as soon as possible. 
+ - **Can I contribute?** Of course! We always welcome extera help! You can fork the Github project, and send us Pull-Request. If you want to discuss the suggested changes/improvment/etc, you can use the issue tracker. 
+ - **Some of the tests that I have not touched failed for some mysterious reasons** Some tests have extra reqirements. For example the [`inference`](inference#frequently-asked-questions) and [`curator`](curator/README.md#frequently-asked-questions) modules. Unless you're changing a very core part of the code-base, you can ignore failures in other modules. If you want to test only a single module you can use `-pl` option. For example:  `mvn -pl pos` tests only the module(s) inside the `pos` folder.
```
Comment (mssammon) [inline faq.md]: I would put this first, before saying "open an issue"

File: inference/README.md [hunk 3]
```diff
@@ -7,13 +7,17 @@ This project is a suite of unified wrappers to a set optimization libraries, as
...
-    export PATH="${PATH}:${GUROBI_HOME}/bin"
-    export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:${GUROBI_HOME}/lib"
+export GUROBI_HOME="PATH-TO-GUROBI/linux64"
+export PATH="${PATH}:${GUROBI_HOME}/bin"
+export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:${GUROBI_HOME}/lib"
 ```
+
+## Frequently Asked Questions 
+
+ - **Why the inference modules tests keep failing?** We have have a unit test for Gurobi which works only when the license is provided in environment. For this reason we skip on Semaphore. If you're running locally and seeing failures it must be that you don't have the license installed on your computer (whcih you can ignore, if you don't need it).
```
Comment (mssammon) [inline inference/README.md]: typo: "whcih" -> "which"

File: faq.md (hunk 2 above)
Comment (mssammon) [inline faq.md]: maybe have a separate section in the FAQ for troubleshooting, which goes first? 
could add a note about Java 8/the```Exception in thread "main" java.lang.UnsupportedClassVersionError: a (Unsupported major.minor version 51.0)```

**Comment by danyaljj** [comment]
Updated. See if you're happy now.

**Comment by danyaljj** [comment]
Since it's approved, merging!

**Pull request merged**